src/main/java/org/candlepin/subscriptions/product/UpstreamProductData.java

```diff
@@ -67,30 +69,27 @@
     PRODUCT_FAMILY,
     PRODUCT_NAME,
     USAGE,
-    /** Originates from opProd attr code "CORES" when coming from a derived product. */
-    X_VIRTUAL_CORES,
-    /** Originates from opProd attr code "IFL" when coming from a derived product. */
-    X_VIRTUAL_IFL,
-    /** Originates from opProd attr code "SOCKET_LIMIT" when coming from a derived product. */
-    X_VIRTUAL_SOCKETS,
-    /** Role originates from opProd field role, not an attribute. */
+    /** Role originates from opProd field roles, not an attribute. */
     X_ROLE;
   }
 
-  /** Maps opProd attribute codes to MidProduct.Attrs of the same name. */
+  /** Maps opProd attribute codes to UpstreamProductData.Attrs of the same name. */
   private static final Map<String, Attr> CODE_TO_ENUM =
       Arrays.stream(Attr.values())
-          .collect(Collectors.toMap(Attr::name, attr -> attr, (a1, a2) -> a1));
+          .collect(Collectors.toUnmodifiableMap(Attr::name, attr -> attr, (a1, a2) -> a1));
 
-  private MidProduct(String sku) {
+  /** Ignore any product attribute whose value is in this set. */
+  private static final Set<String> ATTRIBUTE_DISALLOWED_VALUES = Set.of("n/a", "0", "n", "none");
```

Comment:
I peeked at a few Candlepin pools to see what some of the attributes looked like between a normal pool and a guest derived pool. I noticed a "socket = 0" attribute was missing in the derived sku, so I went back and re-checked the Candlepin Adapter conversion rules. I had forgotten that there is a set of "empty" attribute values which, if Candlepin Adapters encounters, will ignore the attribute outright.

The result is that UpstreamProductData will now also ignore these attributes as well,

Reply:
:+1: